autoload/OmniSharp.vim

```diff
@@ -747,10 +747,18 @@ function! OmniSharp#Install(...) abort
   let l:version = a:000 != [] ? ' -v '.a:000[0] : ''
 
   if has('win32')
-    let l:location = expand('$HOME').'\.omnisharp\omnisharp-roslyn'
-    call system('powershell "& ""'.s:script_location.'""" -H -l "'.l:location
+    if ValidPowerShellSettings()
+      let l:location = expand('$HOME').'\.omnisharp\omnisharp-roslyn'
+      let l:install_result = system('powershell "& ""'.s:script_location.'""" -H -l "'.l:location
           \ .'"'.l:version)
-    echomsg 'OmniSharp installed to: ' . l:location
+      if l:install_result =~# '0'
+        echomsg 'OmniSharp installed to: ' . l:location
+      else
+        echomsg 'Installation to "' . l:location . '" failed inside PowerShell. "'. l:install_result. '" was returned.'
+      endif
+    else
+      echomsg 'Powershell is running at an ExecutionPolicy level that blocks OmniSharp-vim from installing the Roslyn server'
+    endif
```

Comment:
As far as I can tell, the exit status isn't currently being used here - it is still the `Write-Output` value.

Vim has `v:shell_error` which we could use here, it holds the exit status of the last shell command. So we could remove the `Write-Output` lines from the .ps1 script, and do this:

```vim
      call system('powershell "& ""'.s:script_location.'""" -H -l "'.l:location
          \ .'"'.l:version)
      if v:shell_error
        echomsg 'Installation to "' . l:location . '" failed inside PowerShell.'
      else
        echomsg 'OmniSharp installed to: ' . l:location
      endif
```

Reply:
This is a lot cleaner than checking the string! Powershell itself already prints any errors in the console when they occur in the script so the write-outputs are redundant if the vimscript is using the exit codes. In a future update I could see adding some try catch blocks to handle any download or unzipping issues. As well as a flag that removes the zip file after the install.